addons/mail/static/src/discuss/core/public_web/discuss_sidebar_categories.js

```diff
@@ -134,7 +135,14 @@ export class DiscussSidebarChannel extends Component {
     }
 
     get subChannels() {
-        return this.env.filteredThreads?.(this.thread.sub_channel_ids) ?? [];
+        const sortedThreads = (this.env.filteredThreads?.(this.thread.sub_channel_ids) ?? []).sort(
+            threadCompareRegistry.get("mail.last-interest")
+        );
+        const currentThread = this.store.discuss.thread;
+        if (sortedThreads.findIndex((t) => t.eq(currentThread)) > 4) {
+            return [...sortedThreads.slice(0, 4), currentThread];
+        }
+        return sortedThreads.slice(0, 5);
```

Comment:
I feel like it could be simpler, notably by leveraging `notIn` record's method.

```js
        const recentThreads = (this.env.filteredThreads?.(this.thread.sub_channel_ids) ?? [])
            .sort(threadCompareRegistry.get("mail.last-interest"))
            .slice(0, 5);
        if (this.store.discuss.thread?.notIn(recentThreads)) {
            recentThreads[recentThreads.length - 1] = this.store.discuss.thread;
        }
        return recentThreads;